rules/TypeDeclaration/Rector/FunctionLike/AddClosureParamTypeFromArgClassStringRector.php

```diff
@@ -1,201 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Rector\TypeDeclaration\Rector\FunctionLike;
-
-use PhpParser\Node;
-use PhpParser\Node\Arg;
-use PhpParser\Node\Expr\MethodCall;
-use PhpParser\Node\Expr\StaticCall;
-use PhpParser\Node\Identifier;
-use PhpParser\Node\Param;
-use PHPStan\Reflection\ReflectionProvider;
-use PHPStan\Type\Constant\ConstantStringType;
-use PHPStan\Type\Generic\GenericClassStringType;
-use PHPStan\Type\ObjectType;
-use PHPStan\Type\StringType;
-use Rector\Contract\Rector\ConfigurableRectorInterface;
-use Rector\NodeTypeResolver\TypeComparator\TypeComparator;
-use Rector\PHPStanStaticTypeMapper\Enum\TypeKind;
-use Rector\Rector\AbstractRector;
-use Rector\StaticTypeMapper\StaticTypeMapper;
-use Rector\TypeDeclaration\ValueObject\AddClosureParamTypeFromArg;
-use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
-use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
-use Webmozart\Assert\Assert;
-
-/**
- * @see \Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddClosureParamTypeFromArgRector\AddClosureParamTypeFromArgRectorTest
- */
-final class AddClosureParamTypeFromArgClassStringRector extends AbstractRector implements ConfigurableRectorInterface
```

Comment:
We can separate `string` and object type now, so both rule now do the same.  I've merged them together.